src/engine.ts

```diff
@@ -202,18 +210,17 @@ export default class Engine {
         'Wrong player format, expected p1, p2, ...'
       );
       const player = +playerS[1] - 1;
-
-      assert(this.playerToMove() === (player as PlayerEnum), "Wrong turn order in move " + move + ", expected " + this.playerToMove() + ' found ' + player);
+      assert(this.currentPlayer === (player as PlayerEnum), "Wrong turn order in move " + move + ", expected " + this.currentPlayer + ' found ' + player);
 
       const moves = move.substr(2, move.length - 2).trim().split('.');
 
-      for (let i = 0; i < moves.length; i++)  {
-        const split = moves[i].trim().split(' ');
+      for ( const mv of moves )  {
+        const split = mv.trim().split(' ');
         // the final dot is the end turn command
         command = split[0] === "" ? Command.EndTurn : split[0] as Command;
 
-        const available = this.availableCommands;
-        const commandNames = available.map(cmd => cmd.name);
+        this.generateAvailableCommands();
```

Comment:
This seems a bit weird, generating available commands just before reading the move. If they aren't generated beforehand, how can they be transmitted to the UI?